Libraries/Animated/__tests__/Animated-web-test.js

```diff
@@ -0,0 +1,601 @@
+    it('should animate vectors', () => {
+      const vec = new Animated.ValueXY();
+      const vecLayout = vec.getLayout();
+      const opacity = vec.x.interpolate({
+        inputRange: [0, 42],
+        outputRange: [0.2, 0.8],
+      });
+
+      const callback = jest.fn();
+
+      const node = new AnimatedProps(
+        {
+          style: {
+            opacity,
+            transform: vec.getTranslateTransform(),
+            ...vecLayout,
+          },
+        },
+        callback,
+      );
+
+      expect(node.__getValue()).toEqual({
+        style: [
+          {
+            top: vecLayout.top,
+            left: vecLayout.left,
+            opacity,
+            transform: vec.getTranslateTransform(),
+          },
+          {
+            opacity: 0.2,
+            transform: [{translateX: 0}, {translateY: 0}],
+            left: 0,
+            top: 0,
+          },
+        ],
+      });
+
+      node.__attach();
+
+      expect(callback.mock.calls.length).toBe(0);
+
+      vec.setValue({x: 42, y: 1492});
+
+      expect(callback.mock.calls.length).toBe(2); // once each for x, y
+
+      expect(node.__getValue()).toEqual({
+        style: [
+          {
+            top: vecLayout.top,
+            left: vecLayout.left,
+            opacity,
+            transform: vec.getTranslateTransform(),
+          },
+          {
+            opacity: 0.8,
+            transform: [{translateX: 42}, {translateY: 1492}],
+            left: 42,
+            top: 1492,
+          },
+        ],
+      });
+
+      node.__detach();
+
+      vec.setValue({x: 1, y: 1});
+      expect(callback.mock.calls.length).toBe(2);
+    });
+
+    it('should track vectors', () => {
+      const value1 = new Animated.ValueXY();
+      const value2 = new Animated.ValueXY();
+      Animated.timing(value2, {
+        toValue: value1,
+        duration: 0,
+        useNativeDriver: false,
+      }).start();
+      value1.setValue({x: 42, y: 1492});
+      expect(value2.__getValue()).toEqual({x: 42, y: 1492});
+
+      // Make sure tracking keeps working (see stopTogether in ParallelConfig used
+      // by maybeVectorAnim).
+      value1.setValue({x: 3, y: 4});
+      expect(value2.__getValue()).toEqual({x: 3, y: 4});
+    });
+
+    it('should track with springs', () => {
+      const value1 = new Animated.ValueXY();
+      const value2 = new Animated.ValueXY();
+      Animated.spring(value2, {
+        toValue: value1,
+        tension: 3000, // faster spring for faster test
+        friction: 60,
+        useNativeDriver: false,
+      }).start();
+      value1.setValue({x: 1, y: 1});
+      jest.runAllTimers();
+      expect(Math.round(value2.__getValue().x)).toEqual(1);
+      expect(Math.round(value2.__getValue().y)).toEqual(1);
+      value1.setValue({x: 2, y: 2});
+      jest.runAllTimers();
+      expect(Math.round(value2.__getValue().x)).toEqual(2);
+      expect(Math.round(value2.__getValue().y)).toEqual(2);
+    });
```

Comment:
I don't think we need to duplicate all the tests, just the ones related to the expected return value from AnimatedStyle